cachi2/interface/cli.py

```diff
@@ -143,73 +186,66 @@ def fetch_deps(
             "already have a vendor/ directory (will fail if changes would be made)."
         ),
     ),
-    more_flags: str = typer.Option(
-        "",
-        "--flags",
-        help="Pass additional flags as a comma-separated list.",
-        metavar="FLAGS",
-    ),
     log_level: LogLevel = LOG_LEVEL_OPTION,
 ) -> None:
     """Fetch dependencies for supported package managers.
 
     \b
     # gomod package in the current directory
-    cachi2 fetch-deps --package gomod
+    cachi2 fetch-deps gomod
 
     \b
     # pip package (not supported yet) in the root of the source directory
-    cachi2 fetch-deps --source ./my-repo --package pip
+    cachi2 fetch-deps --source ./my-repo pip
 
     \b
     # gomod package in a subpath of the source directory (./my-repo/subpath)
-    cachi2 fetch-deps --source ./my-repo --package '{
+    cachi2 fetch-deps --source ./my-repo '{
         "type": "gomod",
         "path": "subpath"
     }'
 
-    \b
-    # multiple packages
-    cachi2 fetch-deps \\
-        --package gomod \\
-        --package '{"type": "gomod", "path": "subpath"}' \\
-        --package '{"type": "pip", "path": "other-path"}'
-
     \b
     # multiple packages as a JSON list
-    cachi2 fetch-deps --package '[
+    cachi2 fetch-deps '[
         {"type": "gomod"},
         {"type": "gomod", "path": "subpath"},
         {"type": "pip", "path": "other-path"}
     ]'
+
+    \b
+    # multiple packages and flags as a JSON list
+    cachi2 fetch-deps '{
+        "packages": [
+            {"type": "gomod"},
+            {"type": "gomod", "path": "subpath"},
+            {"type": "pip", "path": "other-path"}
+        ],
+        "flags": [
+            "gomod-vendor"
+        ]
+    }'
     """  # noqa: D301, D202; backslashes intentional, blank line required by black
 
-    def parse_packages(package_str: str) -> list[dict]:
-        """Parse a --package argument into a list of packages (--package may be a JSON list)."""
-        if not _looks_like_json(package_str):
-            packages = [{"type": package_str, "path": "."}]
-        elif isinstance(json_obj := json.loads(package_str), dict):
-            packages = [json_obj]
-        else:
-            packages = json_obj
-        return packages
+    input_obj = _Input(input)
 
-    def combine_flags() -> list[str]:
+    def combine_flags(additional_flags: list[str]) -> list[str]:
```

Comment:
Optional: would the naming be clearer like this?
```python
combine_with_cli_flags(json_flags: list[str]) -> list[str]
```
Or `combine_with_option_flags` or something like that